clouddriver-api/src/main/java/com/netflix/spinnaker/clouddriver/orchestration/AnnotationsBasedAtomicOperationsRegistry.java

```diff
@@ -0,0 +1,174 @@
+package com.netflix.spinnaker.clouddriver.orchestration;
+
+import com.google.common.base.Splitter;
+import com.netflix.spinnaker.clouddriver.core.CloudProvider;
+import com.netflix.spinnaker.clouddriver.deploy.DescriptionValidator;
+import com.netflix.spinnaker.clouddriver.exceptions.CloudProviderNotFoundException;
+import com.netflix.spinnaker.kork.exceptions.UserException;
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+// Doesn't seem like logger is being used so removing it
+// @Slf4j
+public class AnnotationsBasedAtomicOperationsRegistry
+    extends ApplicationContextAtomicOperationsRegistry {
+
+  @Autowired List<CloudProvider> cloudProviders;
+
+  // definitely need tests for this method. I had to convert some groovy things to java ehh
+  @Override
+  public AtomicOperationConverter getAtomicOperationConverter(
+      String description, String cloudProvider) {
+    // Legacy naming convention which is not generic and description name is specific to cloud
+    // provider
+    try {
+      AtomicOperationConverter converter =
+          super.getAtomicOperationConverter(description, cloudProvider);
+      // below is the equivalent of Groovys code below?
+      //  if (converter) return converter;
+      if (converter != null) {
+        return converter;
+      }
```

Comment:
since i didnt get as much time as i'd hoped today, i'll just leave this as a comment.....yep! that's how you'd do it in java